.github/workflows/windows.yml

```diff
@@ -28,14 +28,54 @@ jobs:
           environment-name: build_env
           cache-environment: true
           init-shell: cmd.exe
+      - name: Install external dependencies
+        run: micromamba install -n build_env -y -f ./environment-dev.yml
       - name: Build sparrow-ipc
         run: |
           cmake -B build/ -G Ninja ^
             -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} ^
             -DCMAKE_INSTALL_PREFIX=%CONDA_PREFIX% ^
             -DCMAKE_PREFIX_PATH=%CONDA_PREFIX% ^
+            -DCMAKE_CXX_STANDARD=20 ^
             -DSPARROW_IPC_BUILD_SHARED=${{ matrix.build_shared }} ^
-            -DBUILD_TESTS=ON
+            -DSPARROW_IPC_BUILD_TESTS=ON
+          if %errorlevel% neq 0 exit /b %errorlevel%
+          cmake --build build/ --parallel
+          if %errorlevel% neq 0 exit /b %errorlevel%
+      # TODO this is failing (to debug when dependencies can be fetched and built locally with debug mode)
+      #- name: Run tests
+        #run: |
+          #cd build
+          #ctest --output-on-failure
+          #if %errorlevel% neq 0 exit /b %errorlevel%
+
+  windows_build_fetch_from_source:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        build_type: [Debug] # TODO add Release
+        build_shared: [ON, OFF]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Create build environment
+        uses: mamba-org/setup-micromamba@v2
+        with:
+          environment-file: ./environment-dev.yml
+          environment-name: build_env
+          cache-environment: true
+          init-shell: cmd.exe
+      - name: Build sparrow-ipc
+        run: |
+          cmake -B build/ -G Ninja ^
+            -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} ^
+            -DCMAKE_INSTALL_PREFIX=%CONDA_PREFIX% ^
+            -DCMAKE_PREFIX_PATH=%CONDA_PREFIX% ^
+            -DCMAKE_CXX_STANDARD=20 ^
+            -DSPARROW_IPC_BUILD_SHARED=${{ matrix.build_shared }} ^
+            -DFETCH_DEPENDENCIES_WITH_CMAKE=MISSING ^
+            -DCMAKE_VERBOSE_MAKEFILE=ON ^
```

Comment:
I think we can remove this too (was for debugging).

Reply:
fixed